site/blog/top-5-open-source-ai-red-teaming-tools-2025.md

```
@@ -0,0 +1,235 @@
+- Compliance mapping: OWASP, NIST, MITRE, EU AI Act
+- **Best For**: Developers, ML teams, prompt engineers in production workflows
+
+I'd use Promptfoo if I were looking to red team an app in production, especially now that I've become more familiar with enterprise-level red teaming. I prefer to use JS/TS for web projects, and I mainly do web projects, so this is already a no-brainer. Honestly, I'd use it for simpler things too just because of the JS support.
+
+There are evolving features to address the state of the pipeline beyond the LLMs themselves, like MCP testing. MCP might just be a wrapper but I treat giving AI agency with the same level of wariness I'd have of stepping on Lego in the dark an hour after lights-off. If my LLM integration is growing I'll be looking to offload more testing.
+
+### 2. PyRIT (Python Risk Identification Tool)
+
+**Overview**: Hailing from Microsoft, PyRIT works by using converters (for attacks) that are deployed using various strategies.
+
+**Key Features**
```

markdownlint-cli2 (0.17.2)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)